src/main/java/com/matt/forgehax/mods/CoordsFinder.java

```diff
@@ -0,0 +1,191 @@
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+import static com.matt.forgehax.Helper.getFileManager;
+import static com.matt.forgehax.Helper.getLocalPlayer;
+import static com.matt.forgehax.Helper.getWorld;
+import static com.matt.forgehax.Helper.printInform;
+import static java.util.Objects.isNull;
+import static java.util.Objects.nonNull;
+
+@RegisterMod
+public class CoordsFinder extends ToggleMod {
+  @SuppressWarnings("WeakerAccess")
+  public final Setting<Boolean> logLightning =
+      getCommandStub()
+          .builders()
+          .<Boolean>newSettingBuilder()
+          .name("lighting")
+          .description("log lightning strikes")
+          .defaultTo(true)
+          .build();
+
+  @SuppressWarnings("WeakerAccess")
+  public final Setting<Integer> minLightningDist =
+      getCommandStub()
+          .builders()
+          .<Integer>newSettingBuilder()
+          .name("lighting-min-dist")
+          .description("how far a lightning strike has to be from you to get logged")
+          .min(0)
+          .defaultTo(32)
+          .build();
+
+  @SuppressWarnings("WeakerAccess")
+  public final Setting<Boolean> logWolf =
+      getCommandStub()
+          .builders()
+          .<Boolean>newSettingBuilder()
+          .name("wolf")
+          .description("log wolf teleports")
+          .defaultTo(true)
+          .build();
+
+  @SuppressWarnings("WeakerAccess")
+  public final Setting<Integer> minWolfDist =
+      getCommandStub()
+          .builders()
+          .<Integer>newSettingBuilder()
+          .name("wolf-min-dist")
+          .description("how far a wolf teleport has to be from you to get logged")
+          .min(0)
+          .defaultTo(256)
+          .build();
+
+  @SuppressWarnings("WeakerAccess")
+  public final Setting<Boolean> logPlayer =
+      getCommandStub()
+          .builders()
+          .<Boolean>newSettingBuilder()
+          .name("player")
+          .description("log player teleports")
+          .defaultTo(true)
+          .build();
+
+  @SuppressWarnings("WeakerAccess")
+  public final Setting<Integer> minPlayerDist =
+      getCommandStub()
+          .builders()
+          .<Integer>newSettingBuilder()
+          .name("player-min-dist")
+          .description("how far a player teleport has to be from you to get logged")
+          .min(0)
+          .defaultTo(256)
+          .build();
+
+  private final Path log = getFileManager().getBaseResolve("logs/coordsfinder.log");
+  private final SimpleDateFormat timeFormat = new SimpleDateFormat("HH:mm:ss.SSS");
+  private BufferedWriter logWriter;
+
+  public CoordsFinder() {
+    super(Category.MISC, "CoordsFinder", false, "Logs coordinates of lightning strikes and teleports");
+  }
+
+  @Override
+  protected void onEnabled() {
+    super.onEnabled();
+
+    try {
+      if (!Files.exists(log)) Files.createFile(log);
+
+      logWriter = new BufferedWriter(new OutputStreamWriter(new FileOutputStream(log.toFile(), true)));
```

Comment:
You should use `Files.write(path, bytes, StandardOpenOption.CREATE, StandardOpenOption.APPEND)` instead of keeping a handle open to the file. I would recommend doing what you are doing if the write operation was happening extremely frequently, but I don't think its a good idea here.

Reply:
Sorry, I wrote most of this code using PacketLogger as a reference